api/controllers/notifications_controller.py

```diff
@@ -33,34 +50,59 @@ def ack(user_id: int, notification_id: int) -> dict:
 # Admin actions
 @cross_origin()
 @admin_required
-def get_all(_, notification_id: int) -> dict:
+def get_all(_) -> dict:
     """Fetch all notifications"""
-    # TODO: Fetch all notifiations on the system
-    # allow quer string params for pagination and filters
-    return {'status': 'OK', 'notification_id': notification_id}, 200
+    page = request.args.get('p', default=1, type=int)
+    limit = request.args.get('limit', default=10, type=int)
+    if limit > 50 and page > 0:
+        return {'error': 'Limit must be less then 51 and page greater then 0'}
+
+    offset = (page - 1) * limit
+    notif_list = Notifications.select(
+        limit=limit,
+        offset=offset,
+        order_by='created_at DESC',
+    )
+
+    return jsonify(notif_list), 200
 
 @cross_origin()
 @admin_required
 def get_notif_details(_, notification_id: int) -> dict:
     """Get notification information"""
-    # TODO: get notifications data
-    return {'status': 'OK', 'id': notification_id}, 200
+    notif_details = Notifications.select_first(
+        id = notification_id
+    )
+
+    if notif_details is None:
+        return {'error': "Notification doesn't exist"}, 401
+
+    return jsonify(notif_details), 200
 
 @cross_origin()
 @admin_required
 def add_notification(_) -> dict:
     """Add new notification"""
-    # TODO: get data provided via json in the request body and
-    # add new notification
-    return {'status': 'OK'}, 201
+    body = json.loads(request.data)
+
+    try:
+        new_notif = Notifications.insert(**body)
+        return {"Success! ID": new_notif}, 200
```

Comment:
Why is the tag for the notification, "Success! ID" ? Why not just:
```suggestion
        return jsonify(new_notif), 201
```

Check out the status codes definition: https://www.restapitutorial.com/httpstatuscodes.html

this function should return 201, not 200